src/ng/directive/ngModel.js

```diff
@@ -1311,7 +1309,7 @@ function addSetValidityMethod(context) {
     }
 
     toggleValidationCss(validationErrorKey, combinedState);
-    parentForm.$setValidity(validationErrorKey, combinedState, ctrl);
+    ctrl.$$parentForm.$setValidity(validationErrorKey, combinedState, ctrl);
```

Comment:
I think a similar change is needed in [`$setDirty()`][1]: `ctrl.$$parentForm.$setDirty()`

[1]: https://github.com/Narretz/angular.js/blob/fix-parent-form/src/ng/directive/form.js#L222